code/client/src/pages/List.jsx

```diff
@@ -126,8 +128,54 @@ const List = () => {
       w.address !== ONEConstants.EmptyAddress
   }
 
+  const getBase64 = (data) => new Promise((resolve) => {
+    const reader = new FileReader()
+    reader.addEventListener('load', () => resolve(reader.result))
+    reader.readAsText(data)
+  })
+
+  const handleImport = async (info) => {
+    if (info.file.status === 'uploading') {
+      setFileUploading(true)
+    }
+
+    if (info.file.status === 'done') {
+      try {
+        const walletBase64 = await getBase64(info.file.originFileObj)
+        const wallet = JSON.parse(walletBase64)
+
+        dispatch(walletActions.updateWallet(wallet))
+        setFileUploading(false)
+
+        message.info('Imported Successfully')
+      } catch (err) {
+        message.error('File data corrupted')
+      }
+    }
+  }
+
+  const beforeUpload = (file) => {
+    const isJson = file.type === 'application/json'
+    if (!isJson) {
+      message.error('You can only upload json file')
+    }
+
+    return isJson
+  }
+
   return (
-    <>
+    <Space direction='vertical'>
+      <Upload
```

Comment:
I suggest making this its own page, and adding a new top level menu item for it, next to Restore. Or put it in an inconspicuous location at Wallets page

Reply:
Will move to a new tab on restore page.

Reply:
I am working on a new Restore page in `#factory` branch. But I agree it makes the most sense to put under Restore page. Let's make sure it is self-contained.